binary_search_tree/binary_search_tree.py

```diff
@@ -5,13 +5,38 @@ def __init__(self, value):
     self.right = None
 
   def insert(self, value):
-    pass
+    if value == self.value:
+      return "Value found"
+    if value < self.value:
+      if self.left is None:
+        self.left = BinarySearchTree(value)
+      else:
+        self.left.insert(value)
+    elif value > self.value:
+      if self.right is None:
+        self.right = BinarySearchTree(value)
+      else:
+        self.right.insert(value)
+    else:
+      self.value = value
 
   def contains(self, target):
-    pass
+    if self.value == target:
+      return True
+    else:
+      if target < self.value:
+        self.left = BinarySearchTree(self.left)
+        self.left.contains(target)
+      else:
+        self.right = BinarySearchTree(self.right)
+        self.right.contains(target)
 
   def get_max(self):
-    pass
+    if self.right is None:
+      return self.value
+    else:
+      self.right = BinarySearchTree(self.right)
+      self.right.get_max()
 
   def for_each(self, cb):
     pass
```

Comment:
there are a couple of ways to implement this method. One is with recursion. You would need to utilize the cb and then two if statements. 

Another is an iterative approach. you would want to utilize a stack